// task-management/bundles/task-management.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/forms'), require('@angular/material/autocomplete'), require('@angular/material/badge'), require('@angular/material/bottom-sheet'), require('@angular/material/button'), require('@angular/material/button-toggle'), require('@angular/material/card'), require('@angular/material/checkbox'), require('@angular/material/chips'), require('@angular/material/core'), require('@angular/material/datepicker'), require('@angular/material/dialog'), require('@angular/material/divider'), require('@angular/material/expansion'), require('@angular/material/form-field'), require('@angular/material/grid-list'), require('@angular/material/icon'), require('@angular/material/input'), require('@angular/material/list'), require('@angular/material/menu'), require('@angular/material/paginator'), require('@angular/material/progress-bar'), require('@angular/material/progress-spinner'), require('@angular/material/radio'), require('@angular/material/select'), require('@angular/material/sidenav'), require('@angular/material/slide-toggle'), require('@angular/material/slider'), require('@angular/material/snack-bar'), require('@angular/material/sort'), require('@angular/material/stepper'), require('@angular/material/table'), require('@angular/material/tabs'), require('@angular/material/toolbar'), require('@angular/material/tree')) :
    typeof define === 'function' && define.amd ? define('task-management', ['exports', '@angular/core', '@angular/forms', '@angular/material/autocomplete', '@angular/material/badge', '@angular/material/bottom-sheet', '@angular/material/button', '@angular/material/button-toggle', '@angular/material/card', '@angular/material/checkbox', '@angular/material/chips', '@angular/material/core', '@angular/material/datepicker', '@angular/material/dialog', '@angular/material/divider', '@angular/material/expansion', '@angular/material/form-field', '@angular/material/grid-list', '@angular/material/icon', '@angular/material/input', '@angular/material/list', '@angular/material/menu', '@angular/material/paginator', '@angular/material/progress-bar', '@angular/material/progress-spinner', '@angular/material/radio', '@angular/material/select', '@angular/material/sidenav', '@angular/material/slide-toggle', '@angular/material/slider', '@angular/material/snack-bar', '@angular/material/sort', '@angular/material/stepper', '@angular/material/table', '@angular/material/tabs', '@angular/material/toolbar', '@angular/material/tree'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['task-management'] = {}, global.ng.core, global.ng.forms, global.ng.material.autocomplete, global.ng.material.badge, global.ng.material.bottomSheet, global.ng.material.button, global.ng.material.buttonToggle, global.ng.material.card, global.ng.material.checkbox, global.ng.material.chips, global.ng.material.core, global.ng.material.datepicker, global.ng.material.dialog, global.ng.material.divider, global.ng.material.expansion, global.ng.material.formField, global.ng.material.gridList, global.ng.material.icon, global.ng.material.input, global.ng.material.list, global.ng.material.menu, global.ng.material.paginator, global.ng.material.progressBar, global.ng.material.progressSpinner, global.ng.material.radio, global.ng.material.select, global.ng.material.sidenav, global.ng.material.slideToggle, global.ng.material.slider, global.ng.material.snackBar, global.ng.material.sort, global.ng.material.stepper, global.ng.material.table, global.ng.material.tabs, global.ng.material.toolbar, global.ng.material.tree));
}(this, (function (exports, i0, forms, autocomplete, badge, bottomSheet, button, buttonToggle, card, checkbox, chips, core, datepicker, dialog, divider, expansion, formField, gridList, icon, input, list, menu, paginator, progressBar, progressSpinner, radio, select, sidenav, slideToggle, slider, snackBar, sort, stepper, table, tabs, toolbar, tree) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    var TaskManagementService = /** @class */ (function () {
        function TaskManagementService() {
        }
        return TaskManagementService;
    }());
    TaskManagementService.ɵprov = i0__namespace.ɵɵdefineInjectable({ factory: function TaskManagementService_Factory() { return new TaskManagementService(); }, token: TaskManagementService, providedIn: "root" });
    TaskManagementService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    TaskManagementService.ctorParameters = function () { return []; };

    var TaskManagementComponent = /** @class */ (function () {
        function TaskManagementComponent() {
        }
        TaskManagementComponent.prototype.ngOnInit = function () { };
        return TaskManagementComponent;
    }());
    TaskManagementComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'lib-task-management',
                    template: "<mat-horizontal-stepper\r\n  #stepper\r\n  labelPosition=\"bottom\"\r\n  style=\"background-color: white; height: 95px\"\r\n>\r\n  <mat-step label=\"CS\" state=\"phone\">\r\n    <div >\r\n      <lib-create-task></lib-create-task>\r\n    </div>\r\n  </mat-step>\r\n  <mat-step label=\"AT\" state=\"lock\">\r\n    <div>\r\n      <lib-assign-task></lib-assign-task>\r\n    </div>\r\n  </mat-step>\r\n  <mat-step label=\"RT\" state=\"lock\">\r\n    <div ></div>\r\n  </mat-step>\r\n  <mat-step label=\"RAT\" state=\"lock\">\r\n    <div >\r\n      <lib-re-assign-task></lib-re-assign-task>\r\n    </div>\r\n  </mat-step>\r\n  <mat-step label=\"R\" state=\"lock\">\r\n    <div >\r\n      <lib-create-reminder></lib-create-reminder>\r\n    </div>\r\n  </mat-step>\r\n\r\n  <ng-template matStepperIcon=\"phone\">\r\n    <mat-icon>call_end</mat-icon>\r\n  </ng-template>\r\n  <ng-template matStepperIcon=\"lock\">\r\n    <mat-icon>lock</mat-icon>\r\n  </ng-template>\r\n</mat-horizontal-stepper>\r\n"
                },] }
    ];
    TaskManagementComponent.ctorParameters = function () { return []; };
    TaskManagementComponent.propDecorators = {
        stepper: [{ type: i0.ViewChild, args: ['stepper',] }]
    };

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from) {
        for (var i = 0, il = from.length, j = to.length; i < il; i++, j++)
            to[j] = from[i];
        return to;
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }
    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    var CreateTaskComponent = /** @class */ (function () {
        function CreateTaskComponent() {
        }
        CreateTaskComponent.prototype.ngOnInit = function () {
        };
        return CreateTaskComponent;
    }());
    CreateTaskComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'lib-create-task',
                    template: "<p>create-task works!</p>\n",
                    styles: [""]
                },] }
    ];
    CreateTaskComponent.ctorParameters = function () { return []; };

    var AssignTaskComponent = /** @class */ (function () {
        function AssignTaskComponent() {
        }
        AssignTaskComponent.prototype.ngOnInit = function () {
        };
        return AssignTaskComponent;
    }());
    AssignTaskComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'lib-assign-task',
                    template: "<p>assign-task works!</p>\n",
                    styles: [""]
                },] }
    ];
    AssignTaskComponent.ctorParameters = function () { return []; };

    var ReAssignTaskComponent = /** @class */ (function () {
        function ReAssignTaskComponent() {
        }
        ReAssignTaskComponent.prototype.ngOnInit = function () {
        };
        return ReAssignTaskComponent;
    }());
    ReAssignTaskComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'lib-re-assign-task',
                    template: "<p>re-assign-task works!</p>\n",
                    styles: [""]
                },] }
    ];
    ReAssignTaskComponent.ctorParameters = function () { return []; };

    var CreateReminderComponent = /** @class */ (function () {
        function CreateReminderComponent() {
        }
        CreateReminderComponent.prototype.ngOnInit = function () {
        };
        return CreateReminderComponent;
    }());
    CreateReminderComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'lib-create-reminder',
                    template: "<p>create-reminder works!</p>\n",
                    styles: [""]
                },] }
    ];
    CreateReminderComponent.ctorParameters = function () { return []; };

    var materialModules = [
        select.MatSelectModule,
        formField.MatFormFieldModule,
        datepicker.MatDatepickerModule,
        input.MatInputModule,
        checkbox.MatCheckboxModule,
        card.MatCardModule,
        autocomplete.MatAutocompleteModule,
        forms.FormsModule,
        forms.ReactiveFormsModule,
        icon.MatIconModule,
        badge.MatBadgeModule,
        bottomSheet.MatBottomSheetModule,
        button.MatButtonModule,
        buttonToggle.MatButtonToggleModule,
        chips.MatChipsModule,
        dialog.MatDialogModule,
        divider.MatDividerModule,
        expansion.MatExpansionModule,
        gridList.MatGridListModule,
        list.MatListModule,
        menu.MatMenuModule,
        paginator.MatPaginatorModule,
        progressBar.MatProgressBarModule,
        progressSpinner.MatProgressSpinnerModule,
        radio.MatRadioModule,
        core.MatRippleModule,
        sidenav.MatSidenavModule,
        slideToggle.MatSlideToggleModule,
        slider.MatSliderModule,
        snackBar.MatSnackBarModule,
        sort.MatSortModule,
        stepper.MatStepperModule,
        table.MatTableModule,
        tabs.MatTabsModule,
        toolbar.MatToolbarModule,
        tree.MatTreeModule,
    ];
    var TaskManagementModule = /** @class */ (function () {
        function TaskManagementModule() {
        }
        return TaskManagementModule;
    }());
    TaskManagementModule.decorators = [
        { type: i0.NgModule, args: [{
                    declarations: [
                        TaskManagementComponent,
                        CreateTaskComponent,
                        AssignTaskComponent,
                        ReAssignTaskComponent,
                        CreateReminderComponent,
                    ],
                    imports: __spread(materialModules),
                    exports: [
                        TaskManagementComponent,
                        CreateTaskComponent,
                        AssignTaskComponent,
                        ReAssignTaskComponent,
                        CreateReminderComponent,
                    ],
                },] }
    ];

    /*
     * Public API Surface of task-management
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AssignTaskComponent = AssignTaskComponent;
    exports.CreateReminderComponent = CreateReminderComponent;
    exports.CreateTaskComponent = CreateTaskComponent;
    exports.ReAssignTaskComponent = ReAssignTaskComponent;
    exports.TaskManagementComponent = TaskManagementComponent;
    exports.TaskManagementModule = TaskManagementModule;
    exports.TaskManagementService = TaskManagementService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
